File: 프로그래머스/3/43238. 입국심사/입국심사.cpp
#include <string>
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

long long solution(int n, vector<int> times) {
    long long answer = 0;
    
    long long left = 1;
    long long right = *max_element(times.begin(), times.end()) * (long long) n;

    while(left <= right)
    {
        long long mid = (left+right)/2; long long cnt = 0;
        for(int i = 0; i < times.size(); i++)
        {
            cnt += mid / (long long) times[i];
        }
        
        if(cnt >= n)
        {
            right = mid - 1;
            answer = mid;
        }
        else left = mid + 1;
    }
    
    return answer;
    
}

File: 프로그래머스/3/43238. 입국심사/입국심사.js
function getCnt(time, times){
    return times.reduce((acc,cur)=> {
       return acc + Math.floor(time / cur)
    }, 0)
}
function solution(n, times) {
    let s = 1;
    let e = 1e20;
   
    
    while(s + 1 < e){
        const mid = Math.floor((s+e)/2);
        const cnt = getCnt(mid, times);
        
        if(cnt >= n) {
            e = mid;
        } else {
            s = mid;
        }
    }
    
    return e;
}